### DISCUSSION TITLE
When Scrolling to the Top/Bottom the text disappears.

### DISCUSSION THREAD

**Discussion by HassanTaleb90** [krzyzanowskim/STTextView#58] [Questions]
When pressing Command + Up/Down to scroll, the text disappears.
Tested on MacBook Pro (mid-2014).

https://github.com/user-attachments/assets/ee97d514-f9e8-4411-9bfa-8e1d3ace84b0

**Comment by krzyzanowskim** [comment]
thanks. what version is that?

**Reply by HassanTaleb90** [reply]
Xcode: Version 15.3 (15E204a)
STTextView: 0.1.4
Mac OS: 14.6.1 (23G93)

**Reply by krzyzanowskim** [reply]
please update to STTextView 1.0.0 and try again, if that wont help you can try main branch

**Reply by HassanTaleb90** [reply]
> please update to STTextView 1.0.0 and try again, if that wont help you can try main branch

The same problem persists. I believe the computer’s low efficiency plays a significant role in processing long text slowly, and I think this might be why the text disappeared. I believe that newer computers are about 5 to 7 times faster at processing. Based on that, I suggest trying a text that is 7 times longer, which might reveal the problem.

**Reply by krzyzanowskim** [reply]
thanks for reporting that. I believe that is unfortunate bad performance in textkit2 layout engine when trying to layout whole document - that in necessary to jump to the end of the document position. There’s quite few reports to apple about that and I only gor reply that is supposed to be improved in macOS 15. just few days ago @simonbs posted DTS to apple https://developer.apple.com/forums/thread/761364?answerId=799479022&replyId=799479022&login=true about that very problem.

That is not trivial problem s
to solve unfortunately, but is not dismissed at all. Hopefully we manage to come up with usable workaround to make it more performant

**Reply by HassanTaleb90** [reply]
Please check the answer provided by the Apple DTS engineer.
https://forums.developer.apple.com/forums/thread/761364
